src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1474,6 +1501,24 @@ void DoCancelRequest(int backendProcessId, int backendSecretKey)
             }
         }
 
+        internal void ResetCancellation()
+        {
+            _cancellationRequested = false;
+            _userCancellationRequested = false;
+
+            if (ReadCts.IsCancellationRequested)
```

Comment:
This should be encapsulated inside the wrapper (see https://github.com/npgsql/npgsql/pull/3215/files#r500999156).

Also, if Stop hasn't been caused before this logic (which would likely be a bug), there's a race condition here. That's why I'd prefer some DEBUG asserts inside the wrapper making sure we're doing things in order etc...

/cc @Brar

Reply:
Not sure if you want to check for the scheduled cancellation here, or only inside the wrapper as a DEBUG assert.

Reply:
Just meant inside the wrapper as DEBUG asserts.